main.go

```diff
@@ -78,13 +76,34 @@ func main() {
 	log.Println("End processing")
 }
 
-func createRestoreCommand(key, dump string) string {
-	proto := "*4\r\n$7\r\nRESTORE\r\n"
-	key_proto := "$" + strconv.Itoa(len(key)) + "\r\n" + key + "\r\n"
-	ttl_proto := "$1\r\n0\r\n"
-	dump_proto := "$" + strconv.Itoa(len(dump)) + "\r\n" + dump + "\r\n"
+func processKey(client *redis.Client, writer *bufio.Writer, key string) {
+	dump, err := client.Dump(key).Result()
+	if err != nil {
+		log.Printf("ERROR: couldn't dump key %s: %v", key, err)
+		return
+	}
+
+	ttl, err := client.TTL(key).Result()
+	if err != nil {
+		log.Printf("ERROR: couldn't dump key %s: %v", key, err)
+		return
+	}
+
+	writer.WriteString(createRestoreCommand(key, dump, &ttl))
+}
+
+func createRestoreCommand(key, dump string, ttl *time.Duration) string {
+	// TODO: need to check if seconds contains numbers after ,!!!
+	ttlString := strconv.Itoa(int(ttl.Seconds() * 1000))
+
+	result := restoreCommand
+
+	for _, val := range [3]string{key, ttlString, dump} {
+		append := "$" + strconv.Itoa(len(val)) + "\r\n" + val + "\r\n"
+		result = result + append
```

Comment:
`append` is a go keyword, you should avoid using it.
Why not just `result += "$" + strconv.Itoa(len(val)) + "\r\n" + val + "\r\n"`?